### PR TITLE
Data entity cypress

### PR DESCRIPTION
**Pull request by mmb675**
Add new Data Entity  e2e ,visual screenshot and a11y tests to cypress

## Summary of the changes
Tell us, in a few words, what the changes are.

## Related issue
Tell us the issue number. If suggesting an improvement or component, please discuss it with us in an issue first.

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue. 
- [x] Changes to docs package checked and committed.

### CODE REVIEW

File: packages/react/src/component-tests/IcDataEntity/IcDataEntity.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,841 @@
...
+    </div>);
+    cy.checkHydrated("ic-data-entity");
+    cy.get('[heading="Order details"]').should(BE_VISIBLE);
+    cy.get('[label="Drink"]').should(BE_VISIBLE);
+    cy.get("ic-data-row").should(HAVE_VALUE,"Americano");
+    cy.get("ic-link").should(BE_VISIBLE).should(HAVE_TEXT,"Edit");
+    cy.compareSnapshot("drinkName", DEFAULT_TEST_THRESHOLD);
+  });
+
+  it("milk data e2e and visual testing", () => {
```
Comment (GCHQ-Developer-530) [inline packages/react/src/component-tests/IcDataEntity/IcDataEntity.cy.tsx:110]: I think if you're testing one data row on its own, we maybe only need one test for it as there's a lot of fairly similar visual tests

Comment (GCHQ-Developer-530) [inline packages/react/src/component-tests/IcDataEntity/IcDataEntity.cy.tsx:110]: I think the it statements should be more descriptive as well e.g.
```suggestion
  it("should display with only one data row", () => {
```

Comment (GCHQ-Developer-530) [reply packages/react/src/component-tests/IcDataEntity/IcDataEntity.cy.tsx:110]: This applies to all of the if statements, not just the milk one

**Review by GCHQ-Developer-530 — CHANGES_REQUESTED** [review]
Just some comments from me - alongside this, there seems to be some formatting issues (the last few tests are particularly difficult to read, so should just need to run prettier).
You'll also need to add loki skip to the visual tests in storybook so they're not looking for any reference images.

**Comment by GCHQ-Developer-530** [comment]
I think the threshold needs updating as well because they're failing the cypress tests in the CI build.

**Review by GCHQ-Developer-299 — DISMISSED** [review]

**Review by GCHQ-Developer-530 — DISMISSED** [review]

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**